addons/website/controllers/main.py

```diff
@@ -781,6 +781,23 @@ def get_seo_data(self, res_id, res_model):
 
         return res
 
+    @http.route(['/website/check_can_modify_any'], type='json', auth="user", website=True)
+    def check_can_modify_any(self, res_id, res_model, others):
+        if not request.env.user.has_group('website.group_website_restricted_editor'):
+            raise werkzeug.exceptions.Forbidden()
+        try:
+            record = request.env[res_model].browse(res_id)
+            request.website._check_user_can_modify(record)
+        except AccessError:
+            for other in others:
+                try:
+                    record = request.env[other['res_model']].browse(other['res_id'])
+                    request.website._check_user_can_modify(record)
+                    return
+                except AccessError:
+                    continue
+            raise
```

Comment:
What you think about to only have a list as params, main_object is just a record among the others.


```suggestion
    def check_can_modify_any(self, records):
        if not request.env.user.has_group('website.group_website_restricted_editor'):
            raise werkzeug.exceptions.Forbidden()
        for rec in records:
            try:
                record = request.env[rec['res_model']].browse(rec['res_id'])
                request.website._check_user_can_modify(record)
                return True
            except AccessError:
                continue
        raise
```

Reply:
I was wondering... I did keep it distinct so that the first error remained the raised one - but I could also keep track of it within the loop

Reply:
yes better to keep the error on the main object, you are right...

But in this case as well we consider that we will never have a lot of distinct model to check, and we can just reverse the records params, else as you said reraise the first exception AccessError only